spec/controllers/reservations_controller_spec.rb

```diff
@@ -1,247 +1,189 @@
-            res.save(validate: false)
-          end
-        end
-        it 'uses :upcoming as default filter' do
-          get :index
-          expect(assigns(:reservations_set) - Reservation.upcoming.all)
-            .to be_empty
-        end
+      it 'with respect to session[:filter] first' do
+        trait = :checked_out
+        session_trait = :returned
+        session[:filter] = session_trait
+        get :index, trait => true
+        expect(time_filtered).to have_received(session_trait).at_least(:once)
+        expect(time_filtered).to have_received(trait).at_least(:once)
       end
-
-      context 'who is not an admin' do
-        before(:each) do
-          sign_in @user
-          FILTERS.each do |trait|
-            res = FactoryGirl.build(:valid_reservation, trait,
-                                    reserver: [@user, @admin].sample)
-            res.save(validate: false)
-          end
-        end
-
-        it 'uses :reserved as the default filter' do
-          get :index
-          expect(assigns(:reservations_set) - @user.reservations.reserved)
-            .to be_empty
-        end
-      end
-    end
-
-    it_behaves_like 'inaccessible by banned user' do
-      before { get :index }
-    end
-  end
-
-  describe '#show (GET /reservations/:id)' do
-    before(:each) do
-      @admin_res = FactoryGirl.create(:valid_reservation, reserver: @admin)
-    end
-
-    shared_examples 'can view reservation by patron' do
-      before(:each) { get :show, id: @reservation.id }
-      it { is_expected.to render_template(:show) }
-      it { expect(response).to be_success }
-      it { expect(assigns(:reservation)).to eq @reservation }
-    end
-
-    shared_examples 'can view reservation by admin' do
-      before(:each) { get :show, id: @admin_res.id }
-      it { is_expected.to render_template(:show) }
-      it { expect(response).to be_success }
-      it { expect(assigns(:reservation)).to eq @admin_res }
     end
 
-    shared_examples 'cannot view reservation by admin' do
-      before(:each) { get :show, id: @admin_res.id }
-      include_examples 'cannot access page'
-    end
-
-    context 'when accessed by admin' do
-      before(:each) do
-        sign_in @admin
+    context 'normal user' do
+      let!(:user) do
+        res = spy('Array', starts_on_days: time_filtered)
+        UserMock.new(reservations: res)
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'can view reservation by admin'
-    end
-
-    context 'when accessed by checkout person' do
       before(:each) do
-        sign_in @checkout_person
+        mock_user_sign_in(user)
+        get :index
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'can view reservation by admin'
+      it { is_expected.to respond_with(:success) }
+      it { is_expected.to render_template(:index) }
+      it "only gets the current user's reservations" do
```

Comment:
Hm, doesn't this check that we're getting the user's reservations, but not that we're exclusively doing so?